core/api/src/graphql/shared/types/scalar/username.ts

```diff
@@ -23,6 +24,12 @@ function validUsernameValue(value: string) {
   if (value.match(UsernameRegex)) {
     return value.toLowerCase()
   }
+
+  const checkedPhoneNumber = checkedToPhoneNumber(value)
+  if (!(checkedPhoneNumber instanceof Error)) {
+    return value
```

Comment:
if TS allow it you need to return `checkedPhoneNumber` instead of `value` (probably not for this PR but for the next one we will need to handle phone numbers without + so we will need to add them in checkedToPhoneNumber)